src/orbits/keplerian.jl

```diff
@@ -0,0 +1,314 @@
+            return _KeplerianOrbit(p.aRₛ, p.P, p.incl, p.t₀, p.ecc, p.Ω, p.ω)
+        elseif (:b ∈ params) & (:incl ∉ params)
+            incl = compute_incl(p.aRₛ, p.b, p.ecc, sincos(p.ω))
+            return _KeplerianOrbit(p.aRₛ, p.P, incl, p.t₀, p.ecc, p.Ω, p.ω)
+        else
+            throw(ArgumentError("Either incl or b must be specified"))
+        end
+    else
+        throw(ArgumentError("Please specify either ρₛ or aRₛ"))
+    end
+end
+
+# KeywordCalls.jl
+KeplerianOrbit_KC(nt::NamedTuple{(:ρₛ, :Rₛ, :P, :ecc, :t₀, :incl, :Ω, :ω)}) = _KeplerianOrbit(
+    nt.ρₛ, nt.Rₛ, nt.P, nt.ecc, nt.t₀, nt.incl, nt.Ω, nt.ω
+)
+@kwcall KeplerianOrbit_KC(ρₛ, Rₛ, P, ecc, t₀, incl, Ω, ω)
+
+# KeywordDispatch.jl
+@kwdispatch KeplerianOrbit_KD()
+@kwmethod KeplerianOrbit_KD(;ρₛ, Rₛ, P, ecc, t₀, incl, Ω, ω) = _KeplerianOrbit(
+    ρₛ, Rₛ, P, ecc, t₀, incl, Ω, ω
+)
+
+
+#############
+# Orbit logic
+#############
+# Star density
+compute_ρₛ(aRₛ, P, G_nom) = (3.0 * π / (G_nom * P^2.0)) * aRₛ^3.0
+compute_ρₛ(aRₛ, P, G::typeof(G_unit)) = (3.0 * π / (G * P^2.0)) * aRₛ^3.0
+compute_ρₛ(a, P, Rₛ, G) = compute_ρₛ(aRₛ(a, Rₛ), P, G)
+
+# Semi-major axis / star radius ratio
+compute_aRₛ(a, Rₛ) = a / Rₛ
+compute_aRₛ(ρₛ, P, G_nom) = cbrt(G_nom * P^2.0 * ρₛ / (3.0 * π))
+compute_aRₛ(ρₛ, P, G::typeof(G_unit)) = cbrt(G * P^2.0 * ρₛ / (3.0 * π))
+compute_aRₛ(a, P, Rₛ, G) = aRₛ(compute_ρₛ(a, P, Rₛ, G), P)
+
+# Semi-major axis
+compute_a(aRₛ, Rₛ) = aRₛ * Rₛ
+compute_a(M_tot, P, G) = cbrt(G * M_tot * P^2 / (4.0 * π^2))
+compute_a(ρₛ, P, Rₛ, G) = compute_a(compute_aRₛ(ρₛ, P, G), Rₛ)
+
+# Impact parameter
+function compute_b(aRₛ, sincos_incl, ecc, ω)
+    sin_ω, cos_ω = sincos(ω)
+    incl_factor_inv  = (1.0 - ecc^2.0) / (1.0 + ecc * sin_ω)
+    return aRₛ * sincos_incl[2] * incl_factor_inv
+end
+compute_b(ρₛ, P, G, sincos_incl, ecc, ω) = compute_b(compute_aRₛ(ρₛ, P, G), sincos_incl, ecc, ω)
+
+# Inclination
+function compute_incl(aRₛ, b, ecc, sincosω)
+    return acos((b/aRₛ) * (1.0 + ecc*sincosω[1])/(1.0 - ecc^2))
+end
+compute_incl(ρₛ, P, G, b, ecc, sincosω) = compute_incl(compute_aRₛ(ρₛ, P, G), b, ecc, sincosω)
+
+###########
+# RV params
+###########
+compute_M_tot(a, P, G) = 4.0 * π^2.0 * a^3.0 / (G * P^2.0)
+
+function compute_RV_params(ρₛ, Rₛ, a, P, G; Mₚ = zero(typeof(ρₛ * Rₛ^3.0)))
+    M_tot = compute_M_tot(a, P, G)
+    Mₛ = M_tot - Mₚ
+    aₚ = -(Mₛ / M_tot) * a
+    aₛ = a + aₚ
+    return Mₛ, aₛ, Mₚ, aₚ
+end
+
+function compute_M₀(ecc, ω)
+    sin_ω, cos_ω = sincos(ω)
+    E₀ = 2.0 * atan(√(1.0 - ecc) * cos_ω, √(1.0 + ecc) * (1.0 + sin_ω))
+    M₀ = E₀ - ecc * sin(E₀)
+    return M₀
+end
+
+# Finds the position `r` of the planet along its orbit after rotating
+# through the true anomaly `ν`, then transforms this from the
+# orbital plan to the equatorial plane
+# a_rel: aRₛ, aₛ / Rₛ, or aₚ / Rₛ
+# TODO: consider moving this to a separate orbital calculations package in the future
+function _position(orbit, a_rel, t)
```

Comment:
For terminology, instead of using `a_rel` would `K` be accurate? I see this in the exoplanet source and would clarify the different usages of `_position` by `_star_position`, `_planet_position`, and `relative_position`

Reply:
I'm not sure which part of the source you are referring too, but it was my understanding the K is just the semi-amplitude on their RV data. Here, I am just generalizing the different scaled distances that could be passed to `_position`. We could call it `a_scaled` instead of `a_rel`?

Reply:
Yep you're right, I was looking at an RV method! Maybe a good agnostic term is `separation`

Reply:
Sounds good. Will add as soon as I unbork that force-push I did